scripts/Map.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+
+using Game.Scripts;
+
+using Godot;
+
+
+public partial class Map : Node2D
+{
+    private Map _map;
+    private Core core; // Deklaration des Core-Objekts
+    private Player _player;
+    private Ally _ally;
+    private Health _allyHealth;
+    static double _timeElapsed = 0f;
+    private float _allyHealthInterval = 3f;
+
+    public override void _Ready()
+    {
+        _map = this;
+        core = GetNode<Core>("%Core"); // Initialisierung in _Ready
+        GD.Print(core.Scale); // Ausgabe der Skalierung in der Godot-Konsole
+        _player = GetNode<Player>("%Player");
+        _ally = GetNode<Ally>("%Ally");
+        _allyHealth = _ally.GetNode<Health>("Health");
+    }
+
+    public void _DarknessDamage()
```

Comment:
you dont need to check the health before healing or damaging. Health wont go below 0 or above max health when you call heal or damage